dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/readJdbc.kt

```diff
@@ -0,0 +1,635 @@
+ * @return the DataFrame containing the result of the SQL query.
+ */
+public fun DataFrame.Companion.readSqlQuery(dbConfig: DatabaseConfiguration, sqlQuery: String): AnyFrame {
+    DriverManager.getConnection(dbConfig.url, dbConfig.user, dbConfig.password).use { connection ->
+        return readSqlQuery(connection, sqlQuery, DEFAULT_LIMIT)
+    }
+}
+
+/**
+ * Converts the result of an SQL query to the DataFrame.
+ *
+ * @param [dbConfig] the database configuration to connect to the database, including URL, user, and password.
+ * @param [sqlQuery] the SQL query to execute.
+ * @param [limit] the maximum number of rows to retrieve from the result of the SQL query execution.
+ * @return the DataFrame containing the result of the SQL query.
+ */
+public fun DataFrame.Companion.readSqlQuery(dbConfig: DatabaseConfiguration, sqlQuery: String, limit: Int): AnyFrame {
+    DriverManager.getConnection(dbConfig.url, dbConfig.user, dbConfig.password).use { connection ->
+        return readSqlQuery(connection, sqlQuery, limit)
+    }
+}
+
+/**
+ * Converts the result of an SQL query to the DataFrame.
+ *
+ * @param [connection] the database connection to execute the SQL query.
+ * @param [sqlQuery] the SQL query to execute.
+ * @return the DataFrame containing the result of the SQL query.
+ */
+public fun DataFrame.Companion.readSqlQuery(connection: Connection, sqlQuery: String): AnyFrame {
+    return readSqlQuery(connection, sqlQuery, DEFAULT_LIMIT)
+}
+
+/**
+ * Converts the result of an SQL query to the DataFrame.
+ *
+ * @param [connection] the database connection to execute the SQL query.
+ * @param [sqlQuery] the SQL query to execute.
+ * @param [limit] the maximum number of rows to retrieve from the result of the SQL query execution.
+ * @return the DataFrame containing the result of the SQL query.
+ */
+public fun DataFrame.Companion.readSqlQuery(connection: Connection, sqlQuery: String, limit: Int): AnyFrame {
+    val url = connection.metaData.url
+    val dbType = extractDBTypeFromURL(url)
+
+    var internalSqlQuery = sqlQuery
+    if (limit > 0) internalSqlQuery += " LIMIT $limit"
+
+    logger.debug { "Executing SQL query: $internalSqlQuery" }
+
+    connection.createStatement().use { st ->
+        st.executeQuery(internalSqlQuery).use { rs ->
+            val tableColumns = getTableColumnsMetadata(rs)
+            val data = fetchAndConvertDataFromResultSet(tableColumns, rs, dbType, DEFAULT_LIMIT)
+
+            logger.debug { "SQL query executed successfully. Converting data to DataFrame." }
+
+            return data.toDataFrame()
+        }
+    }
+}
+
+/**
+ * Reads the data from a ResultSet and converts it into a DataFrame.
+ *
+ * @param [resultSet] the ResultSet containing the data to read.
+ * @param [dbType] the type of database that the ResultSet belongs to.
+ * @return the DataFrame generated from the ResultSet data.
+ */
+public fun DataFrame.Companion.readResultSet(resultSet: ResultSet, dbType: DbType): AnyFrame {
```

Comment:
Also extension function on ResultSet might be beneficial :)

Reply:
It's a good idea, but I want to postpone all the extension functions for the SQL objects for the 2nd wave